pkg/math/voteweighted/threshold.go

```diff
@@ -0,0 +1,87 @@
+package voteweighted
+
+import (
+	"math/big"
+
+	"cosmossdk.io/math"
+)
+
+var (
+	Zero       = big.NewInt(0)
+	Two        = big.NewInt(2)
+	OneMillion = big.NewInt(1_000_000)
+)
+
+// SignsDiffer determines whether newPrice and newPrime move in different directions relative to old.
+func SignsDiffer(old *big.Int, newPrice *big.Int, newPrime *big.Int) bool {
+	newSign := (&big.Int{}).Sub(old, newPrice).Sign()
+	newPrimeSign := (&big.Int{}).Sub(old, newPrime).Sign()
+	return newSign != newPrimeSign
+}
+
+// CalculateRelativeTicks computes the ticks of newPrime relative to the tick size of newPrice.
+func CalculateRelativeTicks(old *big.Int, newPrice *big.Int, newPrime *big.Int, ppm *big.Int) *big.Int {
```

Comment:
nit: can we name these oldPrice, aggregatedPrice, validatorPrice (or something similar) to indicate what values these variables correspond to intuitively